youtube_dl/extractor/vtm.py

```diff
@@ -0,0 +1,125 @@
+            'sdk': 'js_6.1',
+            'format': 'json',
+            'loginID': username,
+            'password': password,
+        }
+
+        auth_info = self._download_json(
+            'https://accounts.eu1.gigya.com/accounts.login', None,
+            note='Logging in', errnote='Unable to log in',
+            data=urlencode_postdata(auth_data), fatal=True)
+
+        error_message = auth_info.get('errorDetails')
+        if error_message:
+            raise ExtractorError(
+                'Unable to login: %s' % error_message, expected=True)
+
+        self._uid = auth_info.get('UID')
+        self._uid_signature = auth_info.get('UIDSignature')
+        self._signature_timestamp = auth_info.get('signatureTimestamp')
+
+    def _real_initialize(self):
+        self._login()
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        site_id = mobj.group('site_id')
+
+        webpage = self._download_webpage(url, None)
+
+        # There are two known types of URLs both pointing to the same video.
+        # The type of URL you get depends on how you navigate to the video.
+        # Unfortunately, only one type of URL contains the actual video id:
+        #
+        # http://vtm.be/video/volledige-afleveringen/id/257059160170000
+        #
+        # The video id can also be something like:
+        # "vtm_20170213_VM0678492_vtmwatch"
+        #
+        # The other type of URL looks like:
+        #
+        # http://vtm.be/video?aid=166380
+        #
+        # The 'aid' is not the video id we are looking for. There can also be
+        # a lot of very long parameters in addition to the 'aid' parameter.
+        #
+        # So to work in all cases, we extract the video_id from the webpage
+        # instead of the URL.
+        video_id = self._search_regex(
+            r'\\"vodId\\":\\"(.+?)\\"', webpage, 'video_id', default=None)
+
+        # Remove the string ' - Volledige Afleveringen' from the end of the
+        # title (Dutch for "Complete Episodes")
+        title = self._og_search_title(webpage).replace(
+            ' - Volledige Afleveringen', '')
+
+        description = self._html_search_regex(
+            r'<div[^>]+class="field-item\s+even">\s*<p>(.+?)</p>',
+            webpage, 'description', default=None)
+
+        m3u8_data = {
+            'app_id': 'vtm_watch' if site_id == 'vtm' else 'q2',
+            'user_network': 'vtm-sso',
+            'UID': self._uid,
+            'UIDSignature': self._uid_signature,
+            'signatureTimestamp': self._signature_timestamp,
+        }
+
+        data_url = "http://vod.medialaan.io/api/1.0/item/%s/video?%s" % (
+            video_id, compat_urllib_parse_urlencode(m3u8_data))
+        data = self._download_json(data_url, video_id)
+
+        formats = self._extract_m3u8_formats(
+            data["response"]["uri"], video_id, entry_protocol='m3u8_native',
+            ext='mp4', preference=0, fatal=True, m3u8_id='HLS')
```

Comment:
1. `preference` is pointless.
2. `fatal=True` is the default.
3. Should be lowercase `hls`.